src/main/java/org/springframework/data/redis/connection/ReactiveKeyCommands.java

```diff
@@ -175,6 +174,20 @@ default Mono<Boolean> exists(ByteBuffer key) {
 
 		return exists(Mono.just(new KeyCommand(key))).next().map(BooleanResponse::getOutput);
 	}
+	/**
+	 * Determine if given all {@literal keys} exist.
+	 *
+	 * @param keys must not be {@literal null} or {@literal empty}.
+	 * @return {@link Mono} emitting {@literal true} if all keys exist.
+	 * @see <a href="https://redis.io/commands/exists">Redis Documentation: EXISTS</a>
+	 */
+	default Mono<Boolean> exists(List<ByteBuffer> keys) {
```

Comment:
I created a validation method which is for checking if parameterized all keys exist in redis cache. If one of them does not exist, it returns false.
@kutlueren , please check if this function is what you meant to use.

Reply:
Hey @AnneMayor 

Awesome to see that you added the function which carries out what I suggested. It is actually more straight forward than I initially thought, maybe I got confused in the codebase but nice! I shall try it out as soon as it gets merged.

Reply:
This implementation calls `EXISTS` for each key issuing a command for each element in the list.

Redis' [`EXISTS`](https://redis.io/docs/latest/commands/exists/) command accepts multiple arguments and returns how many of the keys exist. Our interface should use the existing functionality instead of putting another layer on top of commands with a worse performance behavior.

The issue we have here is however that `exists(Publisher<KeyCommand> keys)` and `exists(Publisher<List<KeyCommand>> keys)` or `exists(Publisher<MultiKeyCommand>> keys)` erase to the same type and we cannot simply introduce such an override.

I suggest to implement `Mono<Long> exists(List<ByteBuffer> keys)` directly instead of using `default` methods. Also, `ReactiveRedisOperations` should be extended with `countExistingKeys(…)` similar to `RedisTemplate.countExistingKeys(…)`.